shared/types/bridge.ts

```diff
@@ -1,4 +1,196 @@
+import { BigNumber } from 'bignumber.js';
+import { Hex } from '@metamask/utils';
+import { SwapsTokenObject } from '../constants/swaps';
+import { RequestStatus } from '../constants/bridge';
+
 export type ChainConfiguration = {
   isActiveSrc: boolean;
   isActiveDest: boolean;
 };
+
+export enum BridgeFeatureFlagsKey {
+  EXTENSION_CONFIG = 'extensionConfig',
+}
+
+export type BridgeFeatureFlags = {
+  [BridgeFeatureFlagsKey.EXTENSION_CONFIG]: {
+    refreshRate: number;
+    maxRefreshCount: number;
+    support: boolean;
+    chains: { [chainId: Hex]: ChainConfiguration };
+  };
+};
+
+export type BridgeControllerState = {
+  bridgeState: BridgeState;
+};
+
+export type BridgeState = {
+  bridgeFeatureFlags: BridgeFeatureFlags;
+  srcTokens: Record<string, SwapsTokenObject>;
+  srcTopAssets: { address: string }[];
+  destTokens: Record<string, SwapsTokenObject>;
+  destTopAssets: { address: string }[];
+  quoteRequest: Partial<QuoteRequest>;
+  quotes: (QuoteResponse & L1GasFees)[];
+  quotesInitialLoadTime?: number;
+  quotesLastFetched?: number;
+  quotesLoadingStatus?: RequestStatus;
+  quoteFetchError?: string;
+  quotesRefreshCount: number;
+};
```

Comment:
- Rename `BridgeControllerState` to `BridgeState`
- Define new `BridgeControllerState`